### ISSUE TITLE
Crashed parsing a CS2 demo

### ISSUE THREAD

**Issue by thwonghin** [skadistats/clarity#313]
I get a crash with `SimpleRunner` parsing `heroic-vs-big-m2-nuke.dem` downloaded from https://www.hltv.org/matches/2372345/heroic-vs-big-iem-dallas-2024:

```
Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 16384
	at skadistats.clarity.model.state.ClientFrame.getEntity(ClientFrame.java:24)
	at skadistats.clarity.processor.entities.Entities.processPacketEntities(Entities.java:442)
	at skadistats.clarity.processor.entities.Entities.processAndRunPacketEntities(Entities.java:336)
	at skadistats.clarity.processor.entities.Entities.onPacketEntities(Entities.java:331)
	at skadistats.clarity.event.AbstractInvocationPoint.invoke(AbstractInvocationPoint.java:86)
	at skadistats.clarity.event.Event.raise(Event.java:41)
	at skadistats.clarity.processor.reader.InputSourceProcessor.processEmbedded(InputSourceProcessor.java:245)
	at skadistats.clarity.event.AbstractInvocationPoint.invoke(AbstractInvocationPoint.java:86)
	at skadistats.clarity.event.Event.raise(Event.java:41)
	at skadistats.clarity.processor.reader.InputSourceProcessor.processSource(InputSourceProcessor.java:172)
	at skadistats.clarity.event.AbstractInvocationPoint.invoke(AbstractInvocationPoint.java:86)
	at skadistats.clarity.event.Event.raise(Event.java:41)
	at skadistats.clarity.processor.runner.AbstractFileRunner.initAndRunWith(AbstractFileRunner.java:38)
	at skadistats.clarity.processor.runner.SimpleRunner.runWith(SimpleRunner.java:33)
```

**Comment by BusyLivingDota** [comment]
I also have problems parsing the recent CS2 replays. Old replays of 2 months ago are fine, but can't parse recent replays.

**Comment by spheenik** [comment]
Fixed with `clarity-3.0.6`.
Sorry for taking so long.

**Comment by thwonghin** [comment]
Thanks, the latest release works
